syntax = "proto3";

package keto.proto;

enum TransactionStatus {
    INIT = 0;
    ROUTE = 1;
    PRE_CR_BLOCK = 2;
    POST_CR_BLOCK = 3;
    PRE_PROCESS_BALANCE = 4;
    PROCESS = 5;
    WAIT = 6;
    POST_WAIT = 7;
    PRE_DR_BLOCK = 8;
    POST_DR_BLOCK = 9;
    COMPLETE = 10;
    
    FAILURE = 100;
    REJECT = 200;
    
}

/**
 * The transaction definition
 */
message Transaction {
    bytes transaction_hash = 1;
    bytes transaction_signature = 2;
    bytes activeAccount = 3;
    TransactionStatus status = 4;
    bytes asn1_transaction_message = 5;
}


/**
 * The block definition
 */
message Block {
    bytes block_hash = 1;
    bytes block_parent_hash = 2;
    bytes block_signature = 3;
    bytes asn1_block = 4;
}


/**
 * The change set
 */
message ChangeSet {
    bytes asn1_change_set = 1;
}
